store/store_kv.go

```diff
@@ -0,0 +1,140 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/portworx/kvdb"
+)
+
+const (
+	cloudDriveKey           = "clouddrive/"
+	cloudDriveLockKey       = "_lock"
+	maxWatchErrors          = 5
+	defaultLockTryDuration  = 1 * time.Minute
+	defaultLockHoldDuration = 3 * time.Minute
+)
+
+var (
+	// ErrKvdbNotInitialized is returned when kvdb is not initialized
+	ErrKvdbNotInitialized = errors.New("KVDB is not initialized")
+)
+
+type kvStore struct {
+	k                kvdb.Kvdb
+	keyName          string
+	lockTryDuration  time.Duration
+	lockHoldDuration time.Duration
+}
+
+// NewKVStore returns a Store implementation which is a wrapper over
+// kvdb.
+func NewKVStore(kv kvdb.Kvdb, name string) (Store, error) {
+	if len(name) == 0 {
+		name = cloudDriveKey
+	}
+	return NewKVStoreWithParams(kv, name, 0, 0)
+}
+
+// NewKVStoreWithParams returns a Store implementation which is a wrapper over
+// kvdb.
+func NewKVStoreWithParams(
+	kv kvdb.Kvdb,
+	name string,
+	lockTryDuration time.Duration,
+	lockHoldDuration time.Duration,
+) (Store, error) {
+	kstore := kvStore{}
+	if kv == nil {
+		return nil, ErrKvdbNotInitialized
+	}
+	if lockTryDuration != 0 {
+		kstore.lockTryDuration = lockTryDuration
+	} else {
+		kstore.lockTryDuration = kv.GetLockTryDuration()
+	}
+	if kstore.lockTryDuration == 0 {
+		kstore.lockTryDuration = defaultLockTryDuration
+	}
+
+	if lockHoldDuration != 0 {
+		kstore.lockHoldDuration = lockHoldDuration
+	} else {
+		kstore.lockHoldDuration = kv.GetLockHoldDuration()
+	}
+	if kstore.lockHoldDuration == 0 {
+		kstore.lockHoldDuration = defaultLockHoldDuration
+	}
+
+	kstore.k = kv
+	kstore.keyName = cloudDriveKey + name
+	return &kstore, nil
+}
+
+func (kv *kvStore) Lock(owner string) (*StoreLock, error) {
+	return kv.lockWithKeyHelper(owner, kv.keyName)
+}
+
+func (kv *kvStore) Unlock(storeLock *StoreLock) error {
+	kvp, ok := storeLock.internalLock.(*kvdb.KVPair)
+	if !ok {
+		return fmt.Errorf("Invalid StoreLock provided")
+	}
+	return kv.k.Unlock(kvp)
+}
+
+func (kv *kvStore) LockWithKey(owner, key string) (*StoreLock, error) {
+	fullPath := kv.keyName + "/" + key
```

Comment:
The previous code used to take a lock under `clouddrive/_lock/`. Should we follow the same suite to ensure locks are under a prefix? In this way we don't change the behavior for Lock() API

Reply:
I think thats what you are trying to do. Can we rename this variable to `lockPrefix` ?